### ISSUE TITLE
When I run the chat bot and asking questions from the wiki content, it doesn't reply as in the wiki file

### ISSUE THREAD

**Issue by abc20220327** [OptimalScale/LMFlow#420]
i  finetune data wiki_zh_eval.json on vicuna_7b with lora.
When I run the chat bot and asking questions from the wiki content, it doesn't reply as in the wiki file,
what parameters I need to try? i already try num_train_epochs = 5

![image](https://github.com/OptimalScale/LMFlow/assets/102452590/ae8e9697-2728-4363-9eb3-07682aa79503)

![image](https://github.com/OptimalScale/LMFlow/assets/102452590/e24b5340-0cf5-4f33-9491-a9a2073f4de0)

**Comment by shizhediao** [comment]
Hi,
Good questions.
I think it is hard to balance the memorization and generalization.
In your question, you might want the model to memorize some knowledge. If so, the simplest way maybe to overfit the training data with a lot of training epochs. If you really want the model to memorize them, try to set epoch>20.

But it is not the best way because we need to consider the generalization abilities.

What is the size of your data?

**Comment by abc20220327** [comment]
my data  is dowload from  data/download.sh, 

if [ "$1" = "wiki_zh_eval" -o "$1" = "all" ]; then
        echo "downloading wiki_zh_eval dataset"
        filename='wiki_zh_eval.tar.gz'
        wget ${public_server}/${filename}
        tar zxvf ${filename}
        rm ${filename}
    fi


wiki_zh_eval.json data  are almost 3500  text only data

**Comment by abc20220327** [comment]
How do you achieve a balance between  memorization and generalization.

**Comment by shizhediao** [comment]
that is only a toy dataset. Please try to collect more data. for example, > 100K data would be good.
Currently, we do not pursue the model to memorize all knowledge in exchange for good generalization performance.

**Comment by abc20220327** [comment]
If I have 100k data, how much should I set num_train_epoch and learning rate, please give me some advice，thanks

**Comment by shizhediao** [comment]
If you are using lora training, then epoch could be 3 and the learning rate could be 1e-4

**Comment by abc20220327** [comment]
For testing, I still use wiki_zh_eval.json, set epoch to 20, but the result is still the same, the answer is completely different from the training data

**Comment by shizhediao** [comment]
100k means the number of texts.
I feel that it is not strange to see the answer is completely different. Why do you pursue model memorization? 
You'd better try this model on some downstream tasks like chitchat or some exam.

**Comment by abc20220327** [comment]
I am using Lora training for the first time, so I am worried that it is caused by my settings, so I hope you can help

**Comment by shizhediao** [comment]
The model doesn't need to remember every word, but rather learns the knowledge in the data.

**Comment by abc20220327** [comment]
my question and answer is :
![image](https://github.com/OptimalScale/LMFlow/assets/102452590/ee91ce05-245a-4b73-ba61-d9ffbbb0a510)


wiki data is :

![image](https://github.com/OptimalScale/LMFlow/assets/102452590/c9948390-d28d-48ab-bd1d-869fa6df1b91)

![image](https://github.com/OptimalScale/LMFlow/assets/102452590/93a748d9-efd3-483c-b104-e26f1afd8266)


I think finetune didn't let it learn knowledge, is there something wrong with me ？


deepspeed ${deepspeed_args} \
  examples/finetune.py \
    --model_name_or_path /home/cd/ai/text-generation-webui/models/vicuna-7b-1.1 \
    --dataset_path ${dataset_path} \
    --output_dir ${output_dir} --overwrite_output_dir \
    --num_train_epochs 50 \
    --learning_rate 1e-4 \
    --block_size 512 \
    --per_device_train_batch_size 1\
    --use_lora 1 \
    --lora_r 8 \
    --save_aggregated_lora 0\
    --deepspeed configs/ds_config_zero2.json \
    --bf16 \
    --run_name finetune_with_lora \
    --validation_split_percentage 0 \
    --logging_steps 20 \
    --do_train \
    --ddp_timeout 72000 \
    --save_steps 100000 \
    --dataloader_num_workers 1 \
    | tee ${log_dir}/train.log \
    2> ${log_dir}/train.err

**Comment by abc20220327** [comment]
Is it because vicuna cannot perform lora training?

**Comment by shizhediao** [comment]
vicuna can perform lora training in our framework.
How about tuning a llama model on a large data?

**Comment by abc20220327** [comment]
> vicuna can perform lora training in our framework. How about tuning a llama model on a large data?

Why is my training not working，do you have any suggestion for me

**Comment by ShiXiangXiang123** [comment]
你们可以说中文嘛，靠！我一个大专毕业的哪里看得懂这些。

**Comment by research4pan** [comment]
> > vicuna can perform lora training in our framework. How about tuning a llama model on a large data?
> 
> Why is my training not working，do you have any suggestion for me

lora training requires tuning of hyperparameters. You may try more sets hyperparameters to see if it works.

**Comment by shizhediao** [comment]
This issue has been marked as stale because it has not had recent activity. If you think this still needs to be addressed please feel free to reopen this issue. Thanks
